contracts/HATVaultsRegistry.sol

```diff
@@ -0,0 +1,603 @@
+    GeneralParameters public generalParameters;
+    ITokenLockFactory public immutable tokenLockFactory;
+    // feeSetter sets the withdrawal fee
+    address public feeSetter;
+
+    // the token into which a part of the the bounty will be swapped into
+    IERC20 public immutable HAT;
+
+    // asset => hacker address => amount
+    mapping(address => mapping(address => uint256)) public hackersHatReward;
+    // asset => amount
+    mapping(address => uint256) public governanceHatReward;
+
+    HATBountySplit public defaultHATBountySplit;
+
+    address public defaultArbitrator;
+    uint256 public defaultChallengePeriod;
+    uint256 public defaultChallengeTimeOutPeriod;
+
+    bool public isEmergencyPaused;
+
+    event LogClaim(address indexed _claimer, string _descriptionHash);
+    event SetFeeSetter(address indexed _newFeeSetter);
+    event SetWithdrawRequestParams(
+        uint256 _withdrawRequestPendingPeriod,
+        uint256 _withdrawRequestEnablePeriod
+    );
+    event SetClaimFee(uint256 _fee);
+    event SetWithdrawSafetyPeriod(uint256 _withdrawPeriod, uint256 _safetyPeriod);
+    event SetHatVestingParams(uint256 _duration, uint256 _periods);
+    event SetMaxBountyDelay(uint256 _delay);
+    event SetVaultVisibility(address indexed _vault, bool indexed _visible);
+    event VaultCreated(
+        address indexed _vault,
+        address indexed _asset,
+        address _committee,
+        IRewardController _rewardController,
+        string _descriptionHash,
+        uint256 _maxBounty,
+        HATVault.BountySplit _bountySplit,
+        uint256 _bountyVestingDuration,
+        uint256 _bountyVestingPeriods
+    );
+    event SwapAndSend(
+        address indexed _beneficiary,
+        uint256 _amountSwapped,
+        uint256 _amountReceived,
+        address indexed _tokenLock
+    );
+    event SetDefaultHATBountySplit(HATBountySplit _defaultHATBountySplit);
+    event SetDefaultArbitrator(address indexed _arbitrator);
+    event SetDefaultChallengePeriod(uint256 _challengePeriod);
+    event SetDefaultChallengeTimeOutPeriod(uint256 _challengeTimeOutPeriod);
+    event SetEmergencyPaused(bool _isEmergencyPaused);
+
+
+    /**
+    * @notice initialize -
+    * @param _hatVaultImplementation The hat vault implementation address.
+    * @param _hatGovernance The governance address.
+    * @param _HAT the HAT token address
+    * @param _defaultHATBountySplit The default way to split the hat bounty betweeen
+    * the hacker and the governance
+    *   Each entry is a number between 0 and `HUNDRED_PERCENT`.
+    *   Total splits should be less than `HUNDRED_PERCENT`.
+    * @param _tokenLockFactory Address of the token lock factory to be used
+    * to create a vesting contract for the approved claim reporter.
+    */
+    constructor(
+        address _hatVaultImplementation,
+        address _hatGovernance,
+        address _HAT,
+        HATBountySplit memory _defaultHATBountySplit,
```

Comment:
seems easier UX/more explicit to just have two argument here (`governanceHat` and `hackerHatVested`)

Reply:
basically I am proposing to remove the HATBountySplit struct altogether..

Reply:
Don't have a strong opinion really. Should we do it?

Reply:
yes